iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/Devices.java

```diff
@@ -218,9 +218,19 @@ public DeviceTwinName getDeviceTwinNames()
                         } else {
                             this.deviceTwinClient.updateTwin(device.getTwin().toDeviceTwinDevice());
                             DevicePropertyServiceModel model = new DevicePropertyServiceModel();
-                            model.setTags(new HashSet<String>(device.getTwin().getTags().keySet()));
-                            model.setReported(
-                                new HashSet<String>(device.getTwin().getProperties().getReported().keySet()));
+                            if(device.getTwin() != null){
```

Comment:
Using && instead of a bunch of nested ifs seems neater to me. Also fix spacing between ) and {

Reply:
done